pkg/controller/metrics/provision_underway_collector.go

```diff
@@ -74,6 +76,15 @@ func (cc provisioningUnderwayCollector) Collect(ch chan<- prometheus.Metric) {
 			continue // skip reporting the metric for clusterdeployment until the elapsed time is at least minDuration
 		}
 
+		// Skip logging the metric if conditions in desired state
+		if condition == "Unknown" && len(cd.Status.Conditions) > 0 {
```

Comment:
I would rather see all the logic embedded in `getKnownConditions`, so you can immediately (after L66) do something like:
```go
if condition == something {
	continue
}
```